lib/Db/Item.php

```diff
@@ -602,12 +605,14 @@ public function setUnread(bool $unread): self
 
     public function setUrl(string $url = null): self
     {
-        $url = trim($url);
-        if ((strpos($url, 'http') === 0 || strpos($url, 'magnet') === 0)
-            && $this->url !== $url
-        ) {
-            $this->url = $url;
-            $this->markFieldUpdated('url');
+        if (!is_null($url)){
+            $url = trim($url);
+            if ((strpos($url, 'http') === 0 || strpos($url, 'magnet') === 0)
+                && $this->url !== $url
+            ) {
+                $this->url = $url;
+                $this->markFieldUpdated('url');
+            }
         }
```

Comment:
```suggestion
        if (is_null($url)) {
            return $this;
        }
        $url = trim($url);
        if ((strpos($url, 'http') === 0 || strpos($url, 'magnet') === 0) && $this->url !== $url) {
            $this->url = $url;
            $this->markFieldUpdated('url');
        }
```

Reply:
I think that doesn't work. In both cases we want to return $this